libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -975,19 +1074,36 @@ AP_AHRS_NavEKF::EKFType AP_AHRS_NavEKF::ekf_type(void) const
     switch (type) {
 #if CONFIG_HAL_BOARD == HAL_BOARD_SITL
     case EKFType::SITL:
+        return type;
 #endif
+#if HAL_NAVEKF2_AVAILABLE
     case EKFType::TWO:
+        return type;
+#endif
+#if HAL_NAVEKF3_AVAILABLE
     case EKFType::THREE:
         return type;
+#endif
     case EKFType::NONE:
         if (always_use_EKF()) {
+#if HAL_NAVEKF2_AVAILABLE
             return EKFType::TWO;
+#elif HAL_NAVEKF3_AVAILABLE
+            return EKFType::THREE;
+#endif
         }
         return EKFType::NONE;
     }
     // we can get to here if the user has mis-set AHRS_EKF_TYPE - any
-    // value above 3 will get to here.
+    // value above 3 will get to here.  TWO is returned here for no
+    // better reason than "tradition".
+#if HAL_NAVEKF2_AVAILABLE
     return EKFType::TWO;
+#elif HAL_NAVEKF3_AVAILABLE
+    return EKFType::THREE;
```

Comment:
This means that if you don't have TWO compiled in we will return you THREE on misconfiguration.